cli/commands/catalog/tui/update.go

```diff
@@ -0,0 +1,184 @@
+					m.state = pagerState
+				case key.Matches(msg, m.delegateKeys.scaffold):
+					return m, scaffoldModuleCmd(m, selectedModule)
+				}
+			} else {
+				break
+			}
+
+		case key.Matches(msg, m.listKeys.Quit):
+			// because we're on the first screen, we simply quit at this point
+			return m, tea.Quit
+		}
+	}
+
+	// Handle keyboard and mouse events for the list
+	m.list, cmd = m.list.Update(msg)
+	return m, cmd
+}
+
+func updatePager(msg tea.Msg, m model) (tea.Model, tea.Cmd) {
+	var (
+		cmd  tea.Cmd
+		cmds []tea.Cmd
+	)
+
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		bb, cmd := m.buttonBar.Update(msg)
+		m.buttonBar = bb.(*buttonbar.ButtonBar)
+		if cmd != nil {
+			cmds = append(cmds, cmd)
+		}
+		switch {
+		case key.Matches(msg, m.pagerKeys.Choose):
+			// Choose changes the action depending on the active button
+			if m.activeButton == scaffoldBtn {
+				return m, scaffoldModuleCmd(m, m.selectedModule)
+			} else {
+				if err := browser.OpenURL(m.selectedModule.URL()); err != nil {
+					m.viewport.SetContent(fmt.Sprintf("could not open url in browser: %s. got error: %s", m.releaseNotesURL, err))
+				}
+			}
+
+		case key.Matches(msg, m.pagerKeys.Scaffold):
+			return m, scaffoldModuleCmd(m, m.selectedModule)
+
+		case key.Matches(msg, m.pagerKeys.Quit):
+			// because we're on the second screen, we need to go back
+			m.state = listState
+			return m, nil
+		}
+	case buttonbar.ActiveBtnMsg:
+		m.activeButton = button(msg)
+
+	}
+
+	// Handle keyboard and mouse events in the viewport
+	m.viewport, cmd = m.viewport.Update(msg)
+	cmds = append(cmds, cmd)
+
+	return m, tea.Batch(cmds...)
+}
+
+// Update handles all TUI interactions and implements bubbletea.Model.Update.
+func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	case tea.WindowSizeMsg:
+		h, v := appStyle.GetFrameSize()
+		m.list.SetSize(msg.Width-h, msg.Height-v)
+		m.width = msg.Width
+		m.height = msg.Height
+
+		if !m.ready {
+			// Since this program is using the full size of the viewport we
+			// need to wait until we've received the window dimensions before
+			// we can initialize the viewport. The initial dimensions come in
+			// quickly, though asynchronously, which is why we wait for them
+			// here.
+			m.viewport = viewport.New(msg.Width, msg.Height-v-lipgloss.Height(m.footerView()))
+			m.ready = true
+		} else {
+			m.viewport.Width = msg.Width
+			m.viewport.Height = msg.Height - v - lipgloss.Height(m.footerView())
+		}
+
+	case scaffoldFinishedMsg:
+		if msg.err != nil {
+			tea.Printf("error scaffolding module: %s", msg.err.Error())
+		}
+
+		return m, tea.Quit
+
+	case rendererErrMsg:
+		m.viewport.SetContent(fmt.Sprintf("there was an error rendering markdown: %s", msg.err.Error()))
+		// ensure we show the viewport
+		m.state = pagerState
+	}
+
+	// Hand off the message and model to the appropriate update function for the
+	// appropriate view based on the current state.
+	switch m.state {
```

Comment:
Missing default case

Reply:
This is a bit trickier, and I'd rather avoid it to be honest. As the `Update` eventloop [runs so many times per second](https://github.com/charmbracelet/bubbletea/blob/f611ed97a3a63728987874354424b34d7d8efef6/tea.go#L307), I think it would make sense to let the control flow through to the `return m, nil` statement below. I've noticed a lot of their own examples and production apps mainly omit the `default` case. I suspect this is for performance reasons. Note: I'm not bullish on this and I haven't done any benchmarks. If having it there makes you feel more comfortable/satisfies the linter I'll put something in.